topology/src/main/java/io/camunda/zeebe/topology/TopologyChangeAppliers.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.topology.state.MemberState;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation;
+import io.camunda.zeebe.util.Either;
+import java.util.function.UnaryOperator;
+
+/** */
+public interface TopologyChangeAppliers {
+
+  /**
+   * @return the operation applier for the given operation
+   */
+  OperationApplier getApplier(TopologyChangeOperation operation);
+
+  interface OperationApplier {
+
+    /**
+     * Validate and initialize the operation. Returns an Either with an exception if the operation
+     * is not valid, or a function that updates the state of the member in the cluster topology
+     *
+     * @return an either
+     */
+    Either<Exception, UnaryOperator<MemberState>> init();
+
+    /**
+     * Applies the operation. This can be run asynchronously and should complete the future when the
+     * operation is completed.
+     *
+     * @return the future when the operation is completed.
+     */
+    ActorFuture<UnaryOperator<MemberState>> apply();
```

Comment:
:question:  How does `UnaryOperator<MemberState>` deal with unexpected changes? I.e. what happens if the operation was initialized (e.g. updating state from `ACTIVE` to `LEAVING` but now that the operation is completed, the state is `ACTIVE` again? I guess that this is not possible because were are the only one allowed to make changes to the member state and we only run one operation at a time.
:wrench:  If that's the case, I think it makes sense to document that `apply` is guaranteed to read the result of applying the `init` operation.

Reply:
> I guess that this is not possible because were are the only one allowed to make changes to the member state and we only run one operation at a time.

Yes. That is the assumption.

Reply:
I have updated the documentation.